homeassistant/components/knx/__init__.py

```diff
@@ -149,16 +149,13 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     hass.data[DATA_HASS_CONFIG] = config
     conf: ConfigType | None = config.get(DOMAIN)
 
-    if conf is None:
-        # If we have a config entry, setup is done by that config entry.
-        # If there is no config entry, this should fail.
-        return bool(hass.config_entries.async_entries(DOMAIN))
-
-    conf = dict(conf)
-    hass.data[DATA_KNX_CONFIG] = conf
+    if conf:
+        hass.data[DATA_KNX_CONFIG] = dict(conf)
+    elif not hass.config_entries.async_entries(DOMAIN):
```

Comment:
Do we need to check this here? I did because it has been that way before, but can `async_setup` be called when there is no yaml and no config entry for that integration?

Reply:
No, if there is no yaml nor any config entry it won't load the integration setup.

```python
if conf:
        hass.data[DATA_KNX_CONFIG] = dict(conf)
register_knx_services(hass)
return True
```

Reply:
Thanks!